### PR TITLE
fix: correct footers.yml config path and default-fallback nesting

### PR DESCRIPTION
**Pull request by eleshar**
# Bugfix Pull Request

> This repository enforces changelog, release, and label automation for all PRs and issues.
> See the organisation-wide [Automation Governance & Release Strategy](https://github.com/lightspeedwp/.github/blob/main/docs/AUTOMATION_GOVERNANCE.md) for required rules.

## Linked issues

No tracked issue — discovered while investigating the footer-duplication bug fixed in #3443 (already merged). This PR is the follow-up wiring fix that was flagged as out-of-scope there.

## Context

- Severity/Impact: Medium — no data loss or outage, but every markdown file `meta.agent.js` regenerates gets the wrong footer text (flowery `DEFAULT_FOOTERS` fallback instead of the short professional phrases actually configured).
- Affected versions/environments: `scripts/agents/includes/header-footer.js`, invoked by the `documentation.yml` `regenerate` job on every PR/push touching `docs/**` and related paths.

## Reproduction

- Steps: 1) Add or edit `.github/footers.yml` with custom phrases. 2) Call `getFooterPhrases('docs')` (or run `ensureFooter()` on any file). 3) Observe the returned phrase.
- Expected vs Actual: Expected the phrases from `.github/footers.yml`'s `docs` category (e.g. "Made with ❤️ by the LightSpeed team."). Actual: always the hardcoded `DEFAULT_FOOTERS` fallback text ("Docs signed by 🤖 Copilot for LightSpeedWP – always fresh!", "This page brought to you by the 🦄 Magic Automation Unicorns of LightSpeedWP.", etc.), because the config never loaded.

## Root Cause

Two independent bugs in `scripts/agents/includes/header-footer.js`, both causing `getFooterPhrases()` to silently fall back to `DEFAULT_FOOTERS`:

1. **Wrong config path**: `loadFooterConfig()` looked for `.github/automation/footers.yml`. The real file is at `.github/footers.yml`. The path never resolved, `fs.existsSync()` always returned `false`, so the config never loaded, full stop.
2. **Wrong fallback nesting**: the category-not-found fallback read `config.categories.default.phrases`, but `.github/footers.yml`'s `default:` block is a top-level sibling of `categories:`, not nested inside it. That branch was dead code even after fixing bug 1.

Verified directly: `getFooterPhrases('docs')` and `getFooterPhrases('totally-unknown')` both returned `DEFAULT_FOOTERS` content before the fix, and the correct `.github/footers.yml` content after.

## Fix Summary

- `loadFooterConfig()`: corrected path to `.github/footers.yml`.
- `getFooterPhrases()`: corrected fallback to read `config.default.phrases` (top-level) instead of `config.categories.default.phrases`.
- `header-footer.test.js`: updated both assertions, which were pinned to the buggy `DEFAULT_FOOTERS` output ("Have questions? Ping us on GitHub!..."). With the config now loading correctly, the same seed (`"branching"`) deterministically resolves to `.github/footers.yml`'s `docs` category phrase at index 2 ("Prefer a guided setup? Book a consult.") instead — confirmed via the same seeded-hash function `selectFooter()` uses.

## Verification

- [x] Tests added/updated to cover the bug — the two existing `header-footer.test.js` assertions now assert on real config-driven output instead of the buggy fallback.
- [x] Manual verification steps: called `getFooterPhrases('docs')` and `getFooterPhrases('totally-unknown')` directly, confirmed both now return `.github/footers.yml` content, not `DEFAULT_FOOTERS`.
- [x] Negative/edge cases checked: full `scripts/agents/includes/__tests__` + `meta.agent.test.js` suite run — 332/333 pass; the one failure (`label-utils.test.js`, a `<100ms` performance-timing assertion) is pre-existing flakiness unrelated to this change (that file isn't touched by this diff).

## Risk & Rollback

- Risk level: Medium — this changes footer **text** repo-wide going forward. Any file `meta.agent.js` regenerates after this merges gets the short `.github/footers.yml` phrases instead of the flowery fallback text. That's the intended fix, but it's a visible content change, not just an under-the-hood bug fix. Recommend merging this before the repo-wide duplicate-cleanup pass (tracked separately, scoped in a follow-up conversation) so files are only rewritten once with correct content rather than twice.
- Rollback plan: revert this commit — reverts `loadFooterConfig()` and the fallback nesting to their prior (buggy but stable) behaviour; no data migration involved.

### Fixed

- **Footer Config Wiring Fixed** — `loadFooterConfig()` now reads the real `.github/footers.yml` path, and the category-not-found fallback reads the correct top-level `default` block, so footers use the configured phrases instead of always falling back to `DEFAULT_FOOTERS`.

---

### Checklist (Global DoD / PR)

- [x] All AC met and demonstrated
- [x] Tests added/updated (unit/E2E as appropriate)
- [ ] Accessibility checklist completed (where relevant): N/A — no user-facing UI, this is a Node.js automation script.
- [x] Docs/readme/changelog updated (if user-facing) — changelog entry above.
- [ ] Security checklist completed (where relevant): N/A — no untrusted input, no privileged action, `/security-review` run with no findings (see PR discussion).
- [ ] Code/design reviews approved — pending human review.
- [ ] CI green; linked issues closed; release notes prepared (if shipping) — pending CI.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Footer generation now uses the correct configuration file and fallback text, ensuring generated footers reflect the configured content instead of generic placeholders.

- **Tests**
  - Updated footer-related tests to reflect the corrected configuration behavior and current canonical footer wording.

### CODE REVIEW

**Comment by eleshar** [comment]
@coderabbitai review

File: scripts/agents/includes/__tests__/header-footer.test.js [hunk 1]
```diff
@@ -1,74 +1,69 @@
...
-    ).toBe(true);
+    expect(ensureFooter(filePath, { category: 'docs', seed: 'branching' })).toBe(true);
 
-    const output = fs.readFileSync(filePath, "utf8");
-    const footerMatches =
-      output.match(
-        /Have questions\? Ping us on GitHub! 🐙 Made with 💚 by LightSpeedWP/g,
-      ) || [];
+    const output = fs.readFileSync(filePath, 'utf8');
+    const footerMatches = output.match(/Prefer a guided setup\? Book a consult\./g) || [];
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/agents/includes/__tests__/header-footer.test.js]: Fixed — added `ensureFooter uses the top-level default for an unknown category` test, using the same fixture-based isolation as the process.cwd() fix on the other thread. Verified it fails against the pre-fix code (`config.categories.default`) and passes with the actual fix.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: scripts/agents/includes/__tests__/header-footer.test.js [hunk 2]
```diff
@@ -1,74 +1,69 @@
...
-      "*Have questions? Ping us on GitHub! 🐙 Made with 💚 by LightSpeedWP*",
-    );
-    expect(output).not.toContain("https://lightspeedwp.agency/contact");
+    // loadFooterConfig() reads .github/footers.yml (this test runs with
+    // process.cwd() at the repo root, where that file lives), so the
+    // "docs" category's real phrases are used, not the DEFAULT_FOOTERS
+    // fallback. selectFooter()'s hash deterministically maps seed
+    // "branching" to index 2 of that category's phrase list.
+    const output = fs.readFileSync(filePath, 'utf8');
+    expect(output).toContain('Prefer a guided setup? Book a consult.');
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/agents/includes/__tests__/header-footer.test.js]: Fixed — all 3 tests now use their own fixture directory with a minimal .github/footers.yml, mocked via jest.spyOn(process, 'cwd').mockReturnValue(fixtureDir) per test and restored in afterEach. Assertions now check against the fixture's own phrase text rather than the live repo config, so they no longer depend on process.cwd() or on .github/footers.yml's real content staying unchanged. (The ast-grep path-traversal warning on the same lines is a false positive — filePath is always os.tmpdir()-derived within the test, not external input.)

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
